diff --git a/source/bulkdata/profile.c b/source/bulkdata/profile.c
index f5d9bb1..f86bc21 100644
--- a/source/bulkdata/profile.c
+++ b/source/bulkdata/profile.c
@@ -1858,4 +1858,11 @@ unsigned int getMinThresholdDuration(char *profileName)
     return minThresholdDuration;
 }
 
+#ifdef GTEST_ENABLE
+typedef void (*freeRequestURIparamFunc) (void *);
 
+freeRequestURIparamFunc freeRequestURIparamFuncCallback(void)
+{
+   return freeRequestURIparam;
+}
+#endif
diff --git a/source/test/bulkdata/Makefile.am b/source/test/bulkdata/Makefile.am
index fe22d28..489869e 100644
--- a/source/test/bulkdata/Makefile.am
+++ b/source/test/bulkdata/Makefile.am
@@ -18,10 +18,10 @@
 # limitations under the License.
 ##########################################################################
 
-AM_CXXFLAGS = -fno-permissive -std=c++11 -DPERSIST_LOG_MON_REF -fPIC -DPRIVACYMODES_CONTROL -DBULKDATA 
+AM_CXXFLAGS = -fno-permissive -std=c++11  -DGTEST_ENABLE -DPERSIST_LOG_MON_REF -fPIC -DPRIVACYMODES_CONTROL -DBULKDATA 
 #-DFEATURE_SUPPORT_WEBCONFIG -DDROP_ROOT_PRIV
 
-AM_CFLAGS = -DPERSIST_LOG_MON_REF -fPIC 
+AM_CFLAGS = -DPERSIST_LOG_MON_REF -DGTEST_ENABLE  -fPIC 
 
 AUTOMAKE_OPTIONS = subdir-objects
 
diff --git a/source/test/bulkdata/profileTest.cpp b/source/test/bulkdata/profileTest.cpp
index 3041e7a..5f3d9d7 100644
--- a/source/test/bulkdata/profileTest.cpp
+++ b/source/test/bulkdata/profileTest.cpp
@@ -278,7 +278,19 @@ TEST_F(ProfileTest, getMinThresholdDuration_Failure) {
     EXPECT_CALL(*g_vectorMock, Vector_Size(_)).Times(::testing::AtMost(1)).WillRepeatedly(Return(0));
     EXPECT_EQ(getMinThresholdDuration("profile1"), 0);
 }
+#ifdef GTEST_ENABLE
+extern "C" {
+typedef void (*freeRequestURIparamFunc)(void *);
+freeRequestURIparamFunc freeRequestURIparamFuncCallback(void);
+}
 
+TEST_F(ProfileTest, FreeRequestURIparam_Null) {
+    freeRequestURIparamFunc freeFunc = freeRequestURIparamFuncCallback();
+    ASSERT_NE(freeFunc, nullptr);
+    freeFunc(nullptr);
+}
+
+#endif
 #endif
 
 
